### ISSUE TITLE
DNS Resolution Failing During Configuration

### ISSUE THREAD

**Issue by codygreen** [F5Networks/f5-declarative-onboarding#67]
**Bug Request:**
DO 1.6.0-1 is failing DNS resolution when setting NTP and/or BIG-IQ hosts.  When I remove the NTP configuration is moves on to the BIG-IQ license and fails this.

**Error:** 
Error processing system declaration: Error licensing via BIG-IQ getaddrinfo EAI_AGAIN <fqdn removed>:443

**Environment**:
- Azure, BIG-IP 13.1.1.4
- DO 1.6.0-1

**Troubleshooting:**
BIG-IP is configured for DNS via Azure DHCP:
`f5admin@(localhost)(cfg-sync Standalone)(NO LICENSE)(/Common)(tmos)# list /sys dns name-servers
sys dns {
    name-servers { 168.63.129.16 }
}

[f5admin@localhost:NO LICENSE:Standalone] ~ # nslookup
> server 8.8.8.8
Default server: 8.8.8.8
Address: 8.8.8.8#53
> 1.pool.ntp.org
Server:		8.8.8.8
Address:	8.8.8.8#53

Non-authoritative answer:
Name:	1.pool.ntp.org
Address: 72.30.35.89
Name:	1.pool.ntp.org
Address: 198.199.120.223
Name:	1.pool.ntp.org
Address: 108.61.73.244
Name:	1.pool.ntp.org
Address: 103.105.51.156
`

**Declaration:** Sensitive information replaced with ****
`
{
    "schemaVersion": "1.6.0",
    "class": "Device",
    "async": true,
    "Common": {
        "class": "Tenant",
        "hostname": "***.demo",
        "myLicense": {
            "class": "License",
            "licenseType": "licensePool",
            "bigIqHost": "****",
            "bigIqUsername": "****",
            "bigIqPassword": "****",
            "licensePool": "rs_subscription",
            "skuKeyword1": "AWF",
            "skuKeyword2": "1G",
            "unitOfMeasure": "yearly",
            "reachable": false,
            "hypervisor": "azure"
        },
        "myProvisioning": {
            "class": "Provision",
            "ltm": "nominal",
            "asm": "nominal"
        },
        "ext": {
            "class": "VLAN",
            "tag": 1234,
            "mtu": 1500,
            "interfaces": [{
                "name": "1.1",
                "tagged": false
            }]
        },
        "ext-self": {
            "class": "SelfIp",
            "address": "10.90.2.5/24",
            "vlan": "ext",
            "allowService": "all",
            "trafficGroup": "traffic-group-local-only"
        },
        "ext_gw": {
            "class": "Route",
            "gw": "10.90.2.1",
            "network": "default",
            "mtu": 1500
        }
    }
}
`

**Comment by codygreen** [comment]
I am seeing inconsistency with this issue. Some VMs I deploy don't have this issue while others have this issue on initial deployment.  Of those that fail, starting restnoded will sometimes resolve this but not always.

Test infrastructure is being deployed via Terraform and configured via Ansible - so there should not be any changes in the environment between working and failing deployments.  All deployed BIG-IPs are capable of resolving DNS during troubleshooting sessions.

**Comment by focrensh** [comment]
I hit this in AWS this morning as well.

**Comment by codygreen** [comment]
per @focrensh recommendation, I switched the nameServers value from 8.8.8.8 to the internal Azure resource IP 168.63.129.16.  I'm still seeing this error on ~50% of my deployments.

**Comment by zhaoqin-github** [comment]
@focrensh I also encounter this issue. My steps is:

1. Provision a new BIG-IP VE instance in OpenStack private cloud.

2. Install DO rpm via IControl REST API

3. Deploy a initial configuration via DO API. That configuration includes DNS 8.8.8.8 and NTP 0.pool.ntp.org.

DO API has about 50% possibility to returns "invalid config - rolling back", "Unable to resolve host 1.pool.ntp.org: getaddrinfo EAI_AGAIN"

**Comment by zhaoqin-github** [comment]
My DO version is 1.5.0-11.

One example of the error:
```
{
   "statusCode":424,
   "message":{
      "id":"640bae19-d54a-4a11-b234-7467da24a756",
      "selfLink":"https://localhost/mgmt/shared/declarative-onboarding/task/640bae19-d54a-4a11-b234-7467da24a756",
      "code":424,
      "status":"ROLLING_BACK",
      "message":"invalid config - rolling back",
      "errors":[
         "Unable to resolve host 1.pool.ntp.org: getaddrinfo EAI_AGAIN"
      ],
      "result":{
         "class":"Result",
         "code":424,
         "status":"ROLLING_BACK",
         "message":"invalid config - rolling back",
         "errors":[
            "Unable to resolve host 1.pool.ntp.org: getaddrinfo EAI_AGAIN"
         ]
      },
      "declaration":{
         "schemaVersion":"1.5.0",
         "class":"Device",
         "async":true,
         "label":"Basic onboarding",
         "Common":{
            "class":"Tenant",
            "hostname":"679cb66e-9293-472d-b51a-7a67d1d4d9ec.f5bigip.local",
            "myLicense":{
               "class":"License",
               "licenseType":"licensePool",
               "bigIqHost":"10.250.11.138",
               "bigIqUsername":"admin",
               "licensePool":"newman-pool",
               "reachable":true,
               "bigIpUsername":"admin",
               "overwrite":false,
               "unitOfMeasure":"monthly"
            },
            "myProvision":{
               "class":"Provision",
               "ltm":"nominal",
               "asm":"nominal"
            },
            "myDns":{
               "class":"DNS",
               "nameServers":[
                  "8.8.8.8"
               ],
               "search":[
                  "openstack.local"
               ]
            },
            "myNtp":{
               "class":"NTP",
               "servers":[
                  "0.pool.ntp.org",
                  "1.pool.ntp.org"
               ],
               "timezone":"UTC"
            },
            "vlan-failover1":{
               "class":"VLAN",
               "interfaces":[
                  {
                     "name":"1.3",
                     "tagged":false
                  }
               ]
            },
            "vlan-internal1":{
               "class":"VLAN",
               "interfaces":[
                  {
                     "name":"1.2",
                     "tagged":false
                  }
               ]
            },
            "vlan-external2":{
               "class":"VLAN",
               "interfaces":[
                  {
                     "name":"1.1",
                     "tagged":false
                  }
               ]
            },
            "selfip-failover1":{
               "class":"SelfIp",
               "vlan":"vlan-failover1",
               "address":"10.250.16.131/24",
               "trafficGroup":"traffic-group-local-only",
               "allowService":"default"
            },
            "selfip-internal1":{
               "class":"SelfIp",
               "vlan":"vlan-internal1",
               "address":"10.250.17.136/24",
               "trafficGroup":"traffic-group-local-only",
               "allowService":"default"
            },
            "selfip-external2":{
               "class":"SelfIp",
               "vlan":"vlan-external2",
               "address":"10.250.18.133/24",
               "trafficGroup":"traffic-group-local-only",
               "allowService":"default"
            },
            "route-external2":{
               "class":"Route",
               "gw":"10.250.18.1",
               "network":"default"
            },
            "configsync":{
               "class":"ConfigSync",
               "configsyncIp":"10.250.16.131"
            }
         }
      }
   }
}
```

**Comment by dstokesf5** [comment]
This was fixed in the 1.8.0 release. If you are still having a problem with that release, please reopen this issue or file a new one.
